plugin-modernizer-core/src/main/java/io/jenkins/tools/pluginmodernizer/core/recipes/CreateJenkinsFile.java

```diff
@@ -0,0 +1,154 @@
+                        LOG.debug("Visiting POM {}", sourceFile.getSourcePath());
+
+                        // First pass with ArchetypeCommonFileVisitor
+                        PluginMetadata commonMetadata =
+                                new ArchetypeCommonFileVisitor().reduce(tree, new PluginMetadata());
+
+                        // Second pass with PomResolutionVisitor
+                        PluginMetadata pomMetadata = new PomResolutionVisitor().reduce(tree, commonMetadata);
+                        LOG.debug("POM metadata: {}", JsonUtils.toJson(pomMetadata));
+
+                        // Check if Jenkins version exists in properties
+                        if (pomMetadata.getProperties() != null
+                                && pomMetadata.getProperties().containsKey("jenkins.version")) {
+                            String jenkinsVersion = pomMetadata.getProperties().get("jenkins.version");
+                            LOG.debug("Found Jenkins version: {}", jenkinsVersion);
+                            state.setJenkinsVersion(jenkinsVersion);
+                            return tree;
+                        }
+                    }
+                }
+                return tree;
+            }
+        };
+    }
+
+    @Override
+    public Collection<SourceFile> generate(ConfigState state, ExecutionContext ctx) {
+        if (state.jenkinsfileExists) {
+            LOG.debug("Jenkinsfile exists, skipping generation");
+            return Collections.emptyList();
+        }
+
+        String jenkinsVersion = state.getJenkinsVersion();
+        if (jenkinsVersion == null || jenkinsVersion.isEmpty()) {
+            LOG.warn("No Jenkins version found in pom.xml");
+            return Collections.emptyList();
+        }
+
+        LOG.debug("Generating Jenkinsfile for Jenkins version: {}", jenkinsVersion);
+        List<JDK> supportedJdks = JDK.get(jenkinsVersion);
+
+        if (supportedJdks.isEmpty()) {
+            LOG.warn("No supported JDKs found for Jenkins version: {}", jenkinsVersion);
+            return Collections.emptyList();
+        }
+
+        List<JDK> sortedJdks = supportedJdks.stream()
+                .sorted((j1, j2) -> Integer.compare(j2.getMajor(), j1.getMajor()))
+                .collect(Collectors.toList());
+
+        int highestJdk = sortedJdks.get(0).getMajor();
+        int nextJdk = sortedJdks.size() > 1 ? sortedJdks.get(1).getMajor() : highestJdk;
+
+        String jenkinsfileContent = String.format(JENKINSFILE_TEMPLATE, highestJdk, nextJdk);
+        LOG.debug("Generated Jenkinsfile content with JDK versions: {} and {}", highestJdk, nextJdk);
+
+        CreateTextFile createJenkinsfile = new CreateTextFile(
+                jenkinsfileContent, ArchetypeCommonFile.JENKINSFILE.getPath().toString(), false);
+
+        return createJenkinsfile.generate(new AtomicBoolean(true), ctx);
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor(ConfigState state) {
```

Comment:
Since the recipe only create file there is no need to override the getVisitor to not do anything